components/JoinForm/JoinForm.tsx

```diff
@@ -150,20 +153,54 @@ export default function App() {
   };
 
   async function submitJoinFormToMeshDB(joinFormSubmission: JoinFormValues) {
-    console.debug(JSON.stringify(joinFormSubmission));
-
+    // Before we try anything else, submit to S3 for safety.
+    let record: JoinRecord = Object.assign(
+      structuredClone(joinFormSubmission),
+      {
+        submission_time: new Date().toISOString(),
+        code: "",
+        replayed: 0,
+        install_number: null,
+      },
+    ) as JoinRecord;
+
+    saveJoinRecordToS3(record, joinRecordKey).then((key) => {
+      setJoinRecordKey(key as string);
+    });
```

Comment:
Is there a race condition to consider here? (no `await`)

e.g. if this stalls due to throttling and waiting for retries and the one below with more information goes through on the first try, we might overwrite here